Composer/packages/lib/code-editor/src/lg/hooks/useStringArray.ts

```diff
@@ -16,10 +18,30 @@ const getInitialItems = <T extends ArrayBasedStructuredResponseItem>(
   const templateId = getTemplateId(response);
   const template = lgTemplates?.find(({ name }) => name === templateId);
   return response?.value && template?.body
-    ? template?.body?.replace(/- /g, '').split(/\r?\n/g) || []
+    ? template?.body
+        // Split by non-escaped -
+        // eslint-disable-next-line security/detect-unsafe-regex
+        ?.split(/(?<!\\)- /g)
+        // Ignore empty or newline strings
+        ?.filter((s) => s !== '' && s !== '\n')
+        ?.map((s) => s.replace(/\r?\n$/g, ''))
+        // Remove LG template multiline block symbol
+        ?.map((s) => s.replace(/```/g, '')) || []
```

Comment:
I think a normal `.` will work here instead of `?.` on lines 24 to 29 - since we only reach here in the case that both `response.value` and `template.body` are non-null, then we know that the `split` call will work, and so on down the line. You might need to change the checks on line 20 to expliticly `response.value != null && template.body != null` just to work around the falsey values.

Reply:
good call on ```.```
As for the check, I think it's better to avoid non-strict equality checks (== vs. ===).

Reply:
Most of the time, yes, but `== null` and `!= null` are special cases that also check for `undefined`.